geps/gep-1494/index.md

```diff
@@ -130,20 +130,343 @@ From @ongy, some additional goals to keep in mind:
+  certain requirements - like only allowing the disabling of Auth on `/healthz`
+  endpoints or similar use cases.)
+* Should Policy have a way to describe rules around when it should take effect?
+  That's in addition to the usual hierarchical rules, should the Policy have ways
+  to include or exclude specific matches? This would require agreement in the
+  Policy Attachment spec as well.
+
+All of these changes have costs in complexity and troubleshooting difficulty, so
+it's important to ensure that the design consciously makes these tradeoffs.
+
+In particular, the last two items in the above list seem likely to require a fair
+amount of discussion, and including a Policy in the initial release of this
+seems likely to make this change miss its current release window.
+
+
+### Why Envoy's ext_auth?
+
+#### What is ext_auth?
+
+Envoy's External Authorization filter is a filter that calls out to an authorization
+service to check if the incoming request is authorized or not. Note that, in
+order to check _authorization_, it must also be able to determine _authentication_ - 
+this is one of the reasons why we've chosen this approach.
+
+Envoy's implementation of this filter allows both a
+[gRPC, protobuf API](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto)
+and configuration of a HTTP based API (which, as it's not defined using a
+specification like protobuf, requires more configuration).
+
+The important thing to remember here is that the actual authentication and
+authorization processed are delegated to the authorization service - which is
+why the ext_auth approach allows handling many Auth methods - most of the
+work is performed by external services which implement various methods (like
+Basic Auth, OAuth, JWT validation, etc)
+
+#### Why use it over other options?
+
+The community discussed Auth extensively in-person at Kubecon London in early 2025,
+and got broad agreement from multiple dataplanes that:
+
+* something like ext_auth was a good idea, because it's flexible and allows the
+  implementation of many types of Auth without specific protocol implementation
+  in upstream
+* Envoy's ext_auth protocol has no major problems that would stop us using it
+* Envoy-based implementations mostly already have support for it
+
+At that meeting, those present agreed that ext_auth was a good place to start.
+
+Most non-Envoy dataplanes also already have similar methods, so the maintainers
+of projects using other dataplanes were okay with this idea.
+
+The alternative here would be to add a Filter type _per auth method_, which, given
+the large number of options, could quickly become very complex.
+
+This GEP is, however, explicitly _not_ ruling out the possibility of adding
+specific Filters for specific Auth methods in the future, if users of this API
+find the overhead of running a compatible implementation to be too much.
+
+### API Design
+
+#### Phase 1: Adding a Filter
```

Comment:
Do we expect the filter and policy to be 2 CRDs with shared Go struct, or 1 CRD?

Reply:
As @LiorLieberman says, the filter is included in HTTPRoute, the Policy will include that same go struct. I'll clarify.